python/cudf/cudf/core/column/categorical.py

```diff
@@ -936,6 +946,22 @@ def unary_operator(self, unaryop: str):
         )
 
     def __setitem__(self, key, value):
+        if cudf.utils.dtypes.is_scalar(value):
+            new_values = [value]
+        else:
+            new_values = value
+
+        to_add_categories = cudf.Index(new_values).difference(self.categories)
+
+        if (
+            len(to_add_categories)
+            and not to_add_categories.isna()._values.all()
+        ):
```

Comment:
Here we'd need to allow setting a value in category to `np.nan`.

Reply:
Do we actually need to handle `NaN` or should that be treated similar to every other value where it would be a dictionary entry? This is distinctly different than `null`.

Reply:
Yep, while digging into this I came across this issue: https://github.com/rapidsai/cudf/issues/7446

I'll think about how to avoid having this check and fix the above issue..